src/CoreWCF.Primitives/src/CoreWCF/Channels/TransportSecurityBindingElement.cs

```diff
@@ -0,0 +1,216 @@
+                {
+                    SessionSymmetricTransportSecurityProtocolFactory sessionFactory = new SessionSymmetricTransportSecurityProtocolFactory();
+                    // base.ApplyAuditBehaviorSettings(context, sessionFactory);
+                    sessionFactory.SecurityTokenParameters = scParameters.Clone();
+                    ((SecureConversationSecurityTokenParameters)sessionFactory.SecurityTokenParameters).IssuerBindingContext = issuerBindingContext;
+                    this.EndpointSupportingTokenParameters.Endorsing.RemoveAt(0);
+                    try
+                    {
+                        base.ConfigureProtocolFactory(sessionFactory, credentialsManager, true, issuerBindingContext, context.Binding);
+                    }
+                    finally
+                    {
+                        this.EndpointSupportingTokenParameters.Endorsing.Insert(0, scParameters);
+                    }
+
+                    securityServiceDispatcher.SessionMode = true;
+                    securityServiceDispatcher.SessionServerSettings.InactivityTimeout = this.LocalServiceSettings.InactivityTimeout;
+                    securityServiceDispatcher.SessionServerSettings.KeyRolloverInterval = this.LocalServiceSettings.SessionKeyRolloverInterval;
+                    securityServiceDispatcher.SessionServerSettings.MaximumPendingSessions = this.LocalServiceSettings.MaxPendingSessions;
+                    securityServiceDispatcher.SessionServerSettings.MaximumKeyRenewalInterval = this.LocalServiceSettings.SessionKeyRenewalInterval;
+                    securityServiceDispatcher.SessionServerSettings.TolerateTransportFailures = this.LocalServiceSettings.ReconnectTransportOnFailure;
+                    securityServiceDispatcher.SessionServerSettings.CanRenewSession = scParameters.CanRenewSession;
+                    securityServiceDispatcher.SessionServerSettings.IssuedSecurityTokenParameters = scParameters.Clone();
+                    ((SecureConversationSecurityTokenParameters)securityServiceDispatcher.SessionServerSettings.IssuedSecurityTokenParameters).IssuerBindingContext = issuerBindingContext;
+                    securityServiceDispatcher.SessionServerSettings.SecurityStandardsManager = sessionFactory.StandardsManager;
+                    securityServiceDispatcher.SessionServerSettings.SessionProtocolFactory = sessionFactory;
+                    securityServiceDispatcher.SecurityProtocolFactory = (SecurityProtocolFactory)sessionFactory;
+
+                    // pass in the error handler for handling unknown security sessions - dont do this if the underlying channel is duplex since sending 
+                    // back faults in response to badly secured requests over duplex can result in DoS.
+                    //if (context.BindingParameters != null && context.BindingParameters.Find<IChannelDemuxFailureHandler>() == null
+                    //    && !IsUnderlyingListenerDuplex<TChannel>(context))
```

Comment:
This is important for some scenarios. It needs to be implemented before this feature is considered done. But so does the demuxer too.